src/Application/Features/Genres/Commands/UpdateGenre.cs

```diff
@@ -0,0 +1,94 @@
+﻿using AutoMapper;
+using Carter;
+using FluentValidation;
+using LyricsApp.Application.Common.Filters;
+using LyricsApp.Application.Common.Models;
+using LyricsApp.Application.Domain.Entities;
+using LyricsApp.Application.Features.Tags.Commands;
+using LyricsApp.Application.Infrastructure.Persistence;
+using MediatR;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.EntityFrameworkCore;
+
+namespace LyricsApp.Application.Features.Genres.Commands
+{
+    public class UpdateGenre : ICarterModule
+    {
+
+        public void AddRoutes(IEndpointRouteBuilder app)
+        {
+            app.MapPut("api/genres", async (UpdateGenreCommand command, IMediator mediator) =>
+            {
+                return await mediator.Send(command);
+            })
+            .WithName(nameof(UpdateGenre))
+            .WithTags(nameof(Genre))
+            .Produces(StatusCodes.Status404NotFound)
+            .Produces(
+                StatusCodes.Status200OK,
+                typeof(BasicResponse<GenreResponse?>)
+            )
+            .Produces(StatusCodes.Status500InternalServerError)
+            .ProducesValidationProblem()
+            .AddEndpointFilter<ValidationFilter<UpdateGenreCommand>>();
+
+        }
+    }
+
+    public class UpdateGenreCommand : IRequest<IResult>
+    {
+
+        public string GenreId { get; set; }
```

Comment:
Puedes usar el tipo Guid desde aqui en lugar de string```suggestion
        public string GenreId { get; set; }
```